### PR TITLE
PrerequisiteWriterTest: convert to JUnit 5

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

File: code/src/test/pcgen/persistence/lst/output/prereq/PrerequisiteWriterTest.java [hunk 1]
```diff
@@ -456,127 +422,81 @@ protected void setUp() throws Exception
...
-		{
-			if (!bExceptionThrown)
+			else
 			{
-				if (prereq != null)
-				{
-					System.out.println(prereq.toString());
-				}
-				fail("exception expected but not thrown");
+				assertNotNull(PreParserFactory.getInstance().parse(aPreString));
```
Comment (thpr) [inline code/src/test/pcgen/persistence/lst/output/prereq/PrerequisiteWriterTest.java]: This is flawed - it needs to assign prereq

File: code/src/test/pcgen/persistence/lst/output/prereq/PrerequisiteWriterTest.java [hunk 2]
```diff
@@ -456,127 +422,81 @@ protected void setUp() throws Exception
...
-			fail("Could not parse prereq: '" + aPreString + "'");
-		}
 
-		StringWriter sw = new StringWriter();
-		PrerequisiteWriter writer = new PrerequisiteWriter();
-		try
-		{
+			StringWriter sw = new StringWriter();
+			PrerequisiteWriter writer = new PrerequisiteWriter();
 			writer.write(sw, prereq);
```
Comment (thpr) [inline code/src/test/pcgen/persistence/lst/output/prereq/PrerequisiteWriterTest.java]: Given current code, this seems to unconditionally write a null object?

File: code/src/test/pcgen/persistence/lst/output/prereq/PrerequisiteWriterTest.java [hunk 3]
```diff
@@ -456,127 +422,83 @@ protected void setUp() throws Exception
...
```
Comment (thpr) [inline code/src/test/pcgen/persistence/lst/output/prereq/PrerequisiteWriterTest.java]: We still have a problem.  The return statement that used to be on line 492 used to abort when an exception was expected.  I believe after line 433 assertThrows you may need a continue; or some such to ensure that the same logic is followed.  The main problem is prereq can still be null here, and thus we can still try to write a null prereq.

**Comment by grimreaper** [comment]
This is this now conflicting in a still-mergeable way. Closing and will recreate.

**Pull request closed without merging**